proxy/src/telemetry/metrics/prometheus.rs

```diff
@@ -209,9 +218,41 @@ impl fmt::Display for Metrics {
     }
 }
 
+
+// ===== impl Counter =====
+
 impl fmt::Display for Counter {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.0 as f64)
+        write!(f, "{}", (self.0).0 as f64)
+    }
+}
+
+impl Into<f64> for Counter {
+    fn into(self) -> f64 {
+        (self.0).0 as f64
+    }
+}
+
+impl Into<u64> for Counter {
+    fn into(self) -> u64 {
+        (self.0).0 as u64
+    }
+}
+
+impl ops::AddAssign<u64> for Counter {
+    fn add_assign(&mut self, rhs: u64) {
```

Comment:
> BTW, do we really need the ability to add a Counter to a u64? I thought we just need the ability to sum Counters together.

Ah, good point. Will change the `AddAssign` impl to `AddAssign<Counter>`

Reply:
Nevermind, the ability to add a `u64` to a `Counter` _is_ currently necessary, since the `sum` field in a `Histogram` is `Counter`:

https://github.com/runconduit/conduit/blob/0d8150f1df381c997a922f59dbc91a65a3e10941/proxy/src/telemetry/metrics/latency.rs#L68-L69

https://github.com/runconduit/conduit/blob/0d8150f1df381c997a922f59dbc91a65a3e10941/proxy/src/telemetry/metrics/latency.rs#L117

We could change this either by making the `sum` field back into a `Wrapping<u64>`, or by converting the observed latency to a `Counter` before adding, but fundamentally it's still the underlying behaviour of adding an arbitrary unsigned int value --- WDYT?

Reply:
I was mistaken. When I suggested it should be a `Counter` I thought it was a sum of counters, not a sum of latencies. Now I see that of course `Counter` is the wrong type.

If you change it to `Wrapping<u64>` then please add the same comment that we add to `Counter` explaining that Prometheus models these values using f64 and we're assuming that the value will never be too large to represent with a 52-bit f64 mantissa, and/or a TODO to implement correct 52-bit semantics.